### PR TITLE
chore: integrate build process in release pipeline

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- fixed triggering build job.
- wait for build job to complete.
- refactored to make pipeline idempotent

**Related issue(s)**

- https://github.com/kyma-project/kyma-companion/issues/28

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -115,27 +117,48 @@ jobs:
...
+
+      - name: Install requirements
+        run: |
+          pip install -r ./scripts/python/wait-for-commit-check/requirements.txt
+
+      - name: wait for build
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          GIT_REF: ${{ inputs.name }}
+          GIT_CHECK_RUN_NAME: "build-${{ inputs.name }} / Build image"
```
Comment (marcobebway) [inline .github/workflows/create-release.yml:147]: Can you please put a comment here why we decided to use this naming pattern?

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -144,14 +167,16 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           REPOSITORY: ${{ github.repository }}
         run: |
-          RELEASE_ID=$(./hack/scripts/create_draft_release.sh)
+          ./scripts/shell/create_draft_release.sh
+          RELEASE_ID=$(cat release_id.txt)
+          echo "Release ID: $RELEASE_ID"
           echo "release_id=$RELEASE_ID" >> $GITHUB_OUTPUT
     outputs:
       release_id: ${{ steps.create-draft.outputs.release_id }}
```
Comment (marcobebway) [inline .github/workflows/create-release.yml:175]: Why do we output the release id here?

File: README.md [hunk 3]
```diff
@@ -121,11 +121,11 @@ This sections describes how to develop and run the kyma-companion fastapi app.
 ### Running
 You can start the FastApi application with the following options:
 ```bash
-poetry run fastapi dev src/main.py --port 8000
+poetry run fastapi dev src/run.py --port 8000
 ```
 or, directly with FastApi:
 ```bash
-fastapi dev src/main.py --port 8000
+fastapi dev src/run.py --port 8000
```
Comment (marcobebway) [inline README.md:128]: Why is this change needed?

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (mfaizanse) [reply .github/workflows/create-release.yml:175]: If you are talking about `release_id: ${{ steps.create-draft.outputs.release_id }}`, then this is required way to export output from a step so that other steps can use it.

File: README.md (hunk 3 above)
Comment (mfaizanse) [reply README.md:128]: due to lint issues, it was complaining that there are multiple `main.py` in the project.

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (mfaizanse) [reply .github/workflows/create-release.yml:148]: added

File: README.md (hunk 3 above)
Comment (mfaizanse) [reply README.md]: reverted. It was done by auto lint fix.

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (marcobebway) [reply .github/workflows/create-release.yml:176]: Are there other steps using this value? I could not spot any!

Comment (mfaizanse) [reply .github/workflows/create-release.yml:176]: atm no, but later if we want to auto publish the release, it will be used there.
